### DISCUSSION TITLE
Support Flux, SD3m, Kolors, and HyDiT models on the familiar Fooocus UI

### DISCUSSION THREAD

**Discussion by metercai** [lllyasviel/Fooocus#3548] [Show and tell]
SimpleSDXL是在Fooocus基础上衍生的中文分支版本，通过增强的开放架构，最新版本已经支持Flux、SD3m、Kolors和HyDiT模型，最低6G显存可用。
SimpleSDXL is a Chinese branch version derived from Fooocus. With an enhanced open architecture, the latest version already supports Flux, SD3m, Kolors, and HyDiT models, with a minimum of 6G VRAM available.

SimpleSDXL合版了主分支的v2.5.5，具备中文提示翻译器、OneButtonPrompt和IC-Light等模块，功能是Fooocus的超集，可以是你的另一个选择。
SimpleSDXL has merged with the main branch's v2.5.5, featuring modules such as Chinese prompt translator, OneButtonsPrompt, and IC-Light, which are a superset of Fooocus. It can be another option for you.

借助Fooocus强大的架构和丰硕成果，SimpleSDXL获得了很多人的喜爱，非常感谢 lllyasviel 和 mashb1t 。
With the powerful architecture and fruitful achievements of Fooocus, SimpleSDXL has gained the love of many people, Thanks a lot to lllyasviel and mashb1t.

URL: https://github.com/metercai/SimpleSDXL

**Comment by DavidDragonsage** [comment]
@metercai  - This is exciting news, and thank you for posting!

I was unsure from the link whether 6GB of VRAM was the minimum requirement for running SimpleSDXL or if that is only the case if you want to use Flux or SD3 within SimpleSDXL. It is unfortunate there is no Discussion tab attached to the SimpleSDXL github page.

**Reply by metercai** [reply]
6GB of VRAM is only required when using Flux, SD3m, Kolors, and HyDiT.
GitHub is not a convenient tool for Chinese users, so the Discussion tab was not opened. But I can try to open and use it.

**Reply by DavidDragonsage** [reply]
Thank you for the information!

Yes, it would be great to have a place to go to ask questions. I think you are going to have a lot of new non-Chinese users because SimpleSDXL seems to be the natural successor to mainline Fooocus at this point.

**Comment by DavidDragonsage** [comment]
Has anyone been able to track down the main install file, SimpleSDXL2_win.exe.7z? 
I cannot find it at https://hf-mirror.com/

OK, I figured it out. You have to download the install file from https://github.com/metercai/SimpleSDXL using Chrome. Firefox does not work, at least for me.

**Reply by metercai** [reply]
https://huggingface.co/metercai/SimpleSDXL2/tree/main

The latest version:  SimpleSDXL2_win.exe.7z0820

Before using SimpleSDXL2, it is advisable to download and install the model files in advance.

**Reply by DavidDragonsage** [reply]
@metercai  - Thank you so much for the link!

I was not able to download the support files from the main page because I kept getting an error at mirror.com, but
https://huggingface.co/metercai/SimpleSDXL2/tree/main works great!

**Comment by DavidDragonsage** [comment]
Serious security concerns were identified with this package in late December 2024 and a serious security alert has just been published:
https://github.com/LykosAI/StabilityMatrix/security/advisories/GHSA-qq8j-phpf-c63j

I would strongly recommend that everyone cease using SimpleSDXL immediately. RuinedFooocus is a good alternative for those wanting to run Flux in a Fooocus style setting.

**Reply by metercai** [reply]
Obviously, you do not understand the goals of SimpleSDXL. It was clearly stated when creating this repository.
<img width="322" src="https://github.com/user-attachments/assets/a75ef40a-8fe9-4313-bb48-cbc17e376c1c" />

'Chinese' and 'Cloud' are two core keywords. The sub library simpleai_base of SimpleSDXL is closely related to this.
The serious security alert you submitted mainly involves two points:
1, [src/env_utils.rs#L238](https://github.com/metercai/simpleai_base/blob/7b2938b19c565de838a0b3357ea4f3bc986fdd3d/src/env_utils.rs#L238): This is a log statistics code that understands the environment in which SimpleSDXL runs, making it easy for the program to adapt and without any privacy data.  and the systeminfo data can alert and prevent the use of incompatible graphics cards. If non Chinese users are sensitive to this, I can remove it without affecting the final functionality. As for what blockchain to submit to, it's even more alarmist. Can you tell me what value there is in submitting to the blockchain?
2, the rathole function within simpleai_base: This is a testing feature, similar to Gradio's public link(very slow to use in China). But due to the lack of permission mechanism in rathole crate, it was abandoned, and the subsequent plan is to implement it through libp2p rust.

Our starting point is not to do harm, but to facilitate users to provide stronger capabilities. The simpleai_base package will carry P2P based sharing capabilities, including shared preset packages, shared models, and shared computing power, which can be discussed publicly:  https://github.com/metercai/SimpleSDXL/discussions/94

The DID in simpleai_base is a UID with a self verification mechanism, generated from SystemInfo. And upstream third-party nodes authenticate and record. This DID can be self verified and verified by others, thus establishing a trustworthy UID system that facilitates the establishment of subsequent sharing mechanisms. If there is no third-party verification, it can only be used locally through QR code and cannot participate in the sharing program.

DID is the infrastructure of a distributed sharing mechanism that does not rely on a single node and has loose alliance capabilities. This is very helpful for Chinese users to share models and computing power. The biased AI strategies of Western governments targeting Chinese users are very detrimental to the development of technology.

Obviously, The submitter of the security threat alert is not clear about the purpose and purpose of these codes. Perhaps some code is inappropriate, but they are all open source and willing to accept any supervision and questioning.

We are also aware that the sharing plan will bring controversy, so the new dev version has been in public beta for several months, cautious and not releasing the official version. But now we have a clearer understanding of how to implement the sharing plan.

SimpleSDXL has formed its own community in China, and many features are not implemented by me, but contributed by the community, such as FluxAIO, KolorsAIO, and SD15AIO. AIO = All in One, including  t2i and i2i. They have implemented all the functions of Fooocus (SDXLAIO) through the embedded Comfyd engine and have been enhanced.  

If you do not understand or agree, you can choose not to use SimpleSDXL, but do not tarnish its reputation.

**Reply by iwr-redmond** [reply]
> The submitter of the security threat alert is not clear about the purpose and purpose of these codes.

Hello, I am the reporter. Your comment is to an extent correct, because what I _haven't_ yet found in the SimpleSDXL or simpleai_base code is anything about GPU sharing. As far as I can tell there is cloud -> local VPN code, local -> cloud silent logging code, and extensive cloud <-> local handshake (DID) code, but no code that supervises the sharing of GPU resources - although @DavidDragonsage may have noticed something during his work on FooocusPlus. Can you please draw my attention to the functions that coordinate the remote GPU inference, especially code that allows local users to request access to remote GPUs?

**Reply by metercai** [reply]
The sharing plan is a plan, not yet a reality. The first step is to have DID, then before implementing this sharing plan on libp2p rust.

DID provides a trusted authentication mechanism, while libP2P provides communication across private networks. They are all infrastructure. These are all open source, you can research them yourself.

**Comment by DavidDragonsage** [comment]
Unfortunately, "more suitable for Chinese and Cloud" does not tell users of potential security and privacy issues such as calling home with system information, sharing a mobile phone number over an open http channel or the presence of the rathole, whether used or not. 

No one should have to be a coder to understand the risks. Users need to be informed in plain language about these issues so that they can actually make an informed decision whether or not to use the program.

In fact there should be no outgoing communication at all from the UI unless the user freely chooses whether to opt in to resource sharing or not. In other words, completely secure when in standalone mode, or somewhat less secure when in sharing mode - the user's choice.

And with the public beta, there was initially some leverage applied to users to "opt in" because otherwise some features such as basic Flux support were not available. There has since been some relaxation of this lockout, but even very basic features such as access to the OneButtonPrompt and SuperPrompter settings are still not available to those who do not share their mobile phone numbers.

The fact that there was not open communication on these security and privacy issues from the SimpleSDXL2 development team is what has tarnished the reputation of an otherwise brilliant extension of the Fooocus theme.

**Reply by metercai** [reply]
Open source code itself is a kind of announcement, and the acquisition and submission of system information are also obvious. I cannot profit from the existence of these codes either. 

I know that at the sensitivity of code, it may not be suitable for everyone. I will remove the old legacy code to ensure its cleanliness.

There are language and cultural differences between us, and in some cases, we need to use translation software for communication. I also don't have enough time to reply to the questions on GitHub because the demand from users in China is too strong.

The establishment of DID mechanism is an innovative practice, and there has been sufficient communication with our Chinese users during the public beta process. We are also receiving feedback and suggestions for optimization. It is also being improved step by step. 

You can learn about the features on the dev version now, and I hope you can understand why there is not more energy focused on your needs and concerns.
admin panel:
<img width="390"  src="https://github.com/user-attachments/assets/7ed70f4a-febd-4856-af37-53d5a835032a" />
preset store:
<img width="833"  src="https://github.com/user-attachments/assets/973eb424-83b3-4462-93ee-615f41f36fba" />

Demonstration of Minimalist Mode:
The preset for greeting card
<img width="1256"  src="https://github.com/user-attachments/assets/6e8850e9-b984-4038-8878-4d1d50576eae" />
The preset for change clothes
<img width="1255"  src="https://github.com/user-attachments/assets/eea0ae0b-4cf1-4e3d-9657-fe885dab8452" />

SimpleSDXL is not perfect, but it has been practicing its vision of making AI use simpler. 
We will address privacy concerns as soon as possible, but pragmatically, it can only be a balanced choice

**Comment by DavidDragonsage** [comment]
I am very appreciative of all the work that was put into the SimpleSDXL platform, and in fact this generative UI became a part of my daily life.

At this point at least some current and potential users have been made aware of the security and privacy risks involved and they can make the choice for themselves. I appreciate that you will be working on some of these issues as the development process continues. 👍

**Comment by metercai** [comment]
The testing legacy code that you are concerned about has been completely removed from the simpleai_base package. You can check the source code of the project:  https://github.com/metercai/simpleai_base
And the explanation was also provided for the content of simpleai_base: https://github.com/metercai/simpleai_base/blob/main/README.md
<img width="400" src="https://github.com/user-attachments/assets/0af243cc-fa3f-46a1-a496-220c6624da6c" />

By the way, the latest public beta version has already launched the wavespeed feature. Under the combination of fast mode + wavefeed + reserve vram, Flux and SD3.5 achieve a speed increase of over 50% and no OOM at low graphics memory. You can actually experience it and make your own choices.
fast + wavespeed + reserve-vram in admin panel:
<img width="300" src="https://github.com/user-attachments/assets/5d129305-4967-4f59-b077-d56be0be726a" />
